driver.go

```diff
@@ -4,78 +4,111 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"golang.org/x/oauth2"
+	"net/http"
 	"os"
 	"strconv"
 	"sync"
 
 	"github.com/docker/go-plugins-helpers/volume"
-	"github.com/libgolang/log"
 	"github.com/linode/linodego"
+	log "github.com/sirupsen/logrus"
 )
 
 type linodeVolumeDriver struct {
-	linodeAPI   linodego.Client
-	instanceID  *int
-	region      string
-	linodeLabel *string
-	mutex       *sync.Mutex
+	instanceID   int
+	region       string
+	linodeLabel  string
+	linodeToken  string
+	mutex        *sync.Mutex
+	linodeAPIPtr *linodego.Client
 }
 
 // Constructor
-func newLinodeVolumeDriver(linodeAPI linodego.Client, region string, linodeLabel *string) linodeVolumeDriver {
+func newLinodeVolumeDriver(region string, linodeLabel string, linodeToken string) linodeVolumeDriver {
+
 	driver := linodeVolumeDriver{
-		linodeAPI:   linodeAPI,
+		linodeToken: linodeToken,
 		region:      region,
 		linodeLabel: linodeLabel,
 		mutex:       &sync.Mutex{},
 	}
+	return driver
+}
+
+func (driver *linodeVolumeDriver) linodeAPI() (*linodego.Client, error) {
+
```

Comment:
Instantiate Linodego on the fly to avoid initializing the driver when a LINODE_TOKEN is not provided.